server/odc-service/src/main/java/com/oceanbase/odc/service/iam/JwtService.java

```diff
@@ -0,0 +1,295 @@
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.integration.jdbc.lock.JdbcLockRegistry;
+import org.springframework.stereotype.Component;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTCreator;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.TokenExpiredException;
+import com.auth0.jwt.interfaces.Claim;
+import com.oceanbase.odc.metadb.config.SystemConfigEntity;
+import com.oceanbase.odc.service.config.SystemConfigService;
+import com.oceanbase.odc.service.config.model.Configuration;
+import com.oceanbase.odc.service.encryption.SensitivePropertyHandler;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class JwtService {
+
+    private SensitivePropertyHandler sensitivePropertyHandler;
+    // 过期时间 15分钟
+    @Value("${odc.iam.auth.jwt.expire-time:15*60*1000}")
+    private long expireTime;
+    // 缓冲时间 3分钟
+    @Value("${odc.iam.auth.jwt.buffer-time:3*60*1000}")
+    private long bufferTime;
+    // 私钥
+    private String tokenSecret;
+    private static final long TRY_LOCK_TIMEOUT_SECONDS = 5;
+
+    private static final String LOCK_KEY = "ODC_JWT_SECRET_LOCK_KEY";
+
+    private static final String ENCRYPTION_JWT_KEY_SYSTEM_CONFIG_KEY = "odc.iam.auth.jwt.secret-key";
+
+    private static final int ENCRYPTION_KEY_SIZE = 64;
+
+    /**
+     * 确保分布式下密钥一致性
+     * 
+     * @param tokenSecret
+     * @param systemConfigService
+     * @param jdbcLockRegistry
+     */
+    public JwtService(
+            @Value("${odc.iam.auth.jwt.secret-key:#{null}}") String tokenSecret,
+            @Qualifier("sensitivePropertyHandlerImpl") SensitivePropertyHandler sensitivePropertyHandler,
+            SystemConfigService systemConfigService,
+            JdbcLockRegistry jdbcLockRegistry) {
+        try {
+            this.sensitivePropertyHandler = sensitivePropertyHandler;
+            if (Objects.nonNull(tokenSecret)) {
+                this.tokenSecret = sensitivePropertyHandler.decrypt(tokenSecret);
+                return;
+            }
+            log.info("Try to lock odc jwt secret...");
+            Lock lock = jdbcLockRegistry.obtain(LOCK_KEY);
+            if (lock.tryLock(TRY_LOCK_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                try {
+                    log.info("Successfully acquired the jwt secret lock");
+                    // 生成jwt密钥
+                    this.tokenSecret = UUID.randomUUID().toString();
+                    // 对明文密钥加密
+                    String encrypt = sensitivePropertyHandler.encrypt(this.tokenSecret);
```

Comment:
avoid use sensitivePropertyHandler to encrypt tokenSecret here, think about:
the encypriont key inside sensitivePropertyHandler may chanaged, if it happened, tokenSecret will not available then